util/util.go

```diff
@@ -388,16 +404,52 @@ func referSameObject(a, b metav1.OwnerReference) bool {
 }
 
 // PointsTo returns true if any of the owner references point to the given target
+// Deprecated: Use PointsToObject to cover differences in API version or backup/restore that changed UIDs
 func PointsTo(refs []metav1.OwnerReference, target *metav1.ObjectMeta) bool {
 	for _, ref := range refs {
 		if ref.UID == target.UID {
 			return true
 		}
 	}
+	return false
+}
 
+// PointsToObject returns true if any of the owner references point to the given target
+func PointsToObject(obj metav1.Object, target Object) bool {
+	for _, ref := range obj.GetOwnerReferences() {
+		ref := ref
+		if refersTo(&ref, target) {
+			return true
+		}
+	}
 	return false
 }
 
+type Object interface {
+	metav1.Object
+	GroupVersionKind() schema.GroupVersionKind
+}
```

Comment:
Can we make this interface private? There is https://github.com/kubernetes-sigs/controller-runtime/pull/898 upstream that we might be able to use at some point, although I don't want to expose this interface just yet (and make it an API)